src/sinks/gcp/chronicle_unstructured.rs

```diff
@@ -0,0 +1,526 @@
+    tls::{TlsConfig, TlsSettings},
+};
+
+const NAME: &str = "gcp_chronicle_unstructured";
+
+#[derive(Debug, Snafu)]
+#[snafu(visibility(pub))]
+pub enum GcsHealthcheckError {
+    #[snafu(display("log_type template parse error: {}", source))]
+    LogTypeTemplate { source: TemplateParseError },
+
+    #[snafu(display("Endpoint not found"))]
+    NotFound,
+}
+
+#[derive(Debug, Clone, Copy, Eq, PartialEq, Serialize, Deserialize)]
+#[serde(rename_all = "snake_case")]
+pub enum Region {
+    Eu,
+    Us,
+    Asia,
+}
+
+impl Region {
+    /// Each region has a its own endpoint.
+    const fn endpoint(self) -> &'static str {
+        match self {
+            Region::Eu => "https://europe-malachiteingestion-pa.googleapis.com",
+            Region::Us => "https://malachiteingestion-pa.googleapis.com",
+            Region::Asia => "https://asia-southeast1-malachiteingestion-pa.googleapis.com",
+        }
+    }
+}
+
+#[derive(Deserialize, Serialize, Debug)]
+pub struct ChronicleUnstructuredConfig {
+    pub endpoint: Option<String>,
+    pub region: Option<Region>,
+    pub customer_id: String,
+    #[serde(flatten)]
+    pub auth: GcpAuthConfig,
+    #[serde(default)]
+    pub batch: BatchConfig<BulkSizeBasedDefaultBatchSettings>,
+    pub encoding: EncodingConfig,
+    #[serde(default)]
+    pub request: TowerRequestConfig,
+    #[serde(default)]
+    compression: Compression,
+    pub tls: Option<TlsConfig>,
+    pub log_type: Template,
+    #[serde(
+        default,
+        deserialize_with = "crate::serde::bool_or_struct",
+        skip_serializing_if = "crate::serde::skip_serializing_if_default"
+    )]
+    acknowledgements: AcknowledgementsConfig,
+}
+
+inventory::submit! {
+    SinkDescription::new::<ChronicleUnstructuredConfig>(NAME)
+}
+
+impl GenerateConfig for ChronicleUnstructuredConfig {
+    fn generate_config() -> toml::Value {
+        toml::from_str(indoc! {r#"
+            credentials_path = "/path/to/credentials.json"
+            customer_id = "customer_id"
+            log_type = "log_type"
+            encoding.codec = "text"
+        "#})
+        .unwrap()
+    }
+}
+
+pub fn build_healthcheck(
+    client: HttpClient,
+    base_url: &str,
+    auth: GcpAuthenticator,
+) -> crate::Result<Healthcheck> {
+    let uri = base_url.parse::<Uri>()?;
+
+    let healthcheck = async move {
+        let mut request = http::Request::get(&uri).body(Body::empty())?;
+        auth.apply(&mut request);
+
+        let response = client.send(request).await?;
+        healthcheck_response(response, auth, GcsHealthcheckError::NotFound.into())
+    };
+
+    Ok(Box::pin(healthcheck))
+}
+
+#[derive(Debug, Snafu)]
+pub enum ChronicleError {
+    #[snafu(display("Region or endpoint not defined"))]
+    RegionOrEndpoint,
+    #[snafu(display("You can only specify one of region or endpoint"))]
+    BothRegionAndEndpoint,
+}
+
+#[async_trait::async_trait]
+#[typetag::serde(name = "gcp_chronicle_unstructured")]
```

Comment:
I'm not sure exactly in the code where it gets populated, but I had a typo in my sink config, and the suggestions it gave was:

```
2022-07-20T19:44:24.832042Z ERROR vector::cli: Configuration error. error=unknown variant `chronicle_unstructured`, expected one of `aws_cloudwatch_logs`, `aws_cloudwatch_metrics`, `aws_kinesis_firehose`, `aws_kinesis_streams`, `aws_s3`, `aws_sqs`, `azure_blob`, `azure_monitor_logs`, `blackhole`, `clickhouse`, `console`, `datadog_archives`, `datadog_events`, `datadog_logs`, `datadog_metrics`, `datadog_traces`, `elasticsearch`, `file`, `gcp_cloud_storage`, `gcp_pubsub`, `gcp_stackdriver_logs`, `gcp_stackdriver_metrics`, `honeycomb`, `http`, `humio_logs`, `humio_metrics`, `influxdb_logs`, `influxdb_metrics`, `kafka`, `logdna`, `loki`, `nats`, `new_relic`, `new_relic_logs`, `papertrail`, `prometheus`, `prometheus_exporter`, `prometheus_remote_write`, `pulsar`, `redis`, `sematext_logs`, `sematext_metrics`, `socket`, `splunk_hec`, `splunk_hec_logs`, `splunk_hec_metrics`, `statsd`, `unit_test`, `vector`, `websocket` for key `sinks.sink0`
```

Is it because this sink is missing:

```Rust
inventory::submit! {
     SinkDescription::new::<PubsubConfig>("gcp_chronicle_unstructured")
}
```

?